### PR TITLE
fix some warnings, get rid of unused arguments, make list serializer …

### PR DESCRIPTION
**Pull request by makosblade**
…testable

### CODE REVIEW

**Review by katherinebernstein — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -51,13 +51,7 @@ ext.targetCompatibility = JavaVersion.VERSION_11
 
 tasks.withType(JavaCompile) {
     options.compilerArgs += ["--add-modules", "java.se",
-                             "--add-exports", "java.base/jdk.internal.ref=ALL-UNNAMED",
```
Comment (geekbeast) [inline build.gradle:54]: This looks like just compiler args so might be fine, but it is my understanding that the se are required for optimal operation: https://docs.hazelcast.org/docs/latest/manual/html-single/#running-in-modular-java

**Review by geekbeast — CHANGES_REQUESTED** [review]
compilerArgs might be the wrong option, but removed arguments should be JVM arguments (might be a pre-existing issue).

**Comment by makosblade** [comment]
Ah derp, I'll update and move them to JVM args. They don't do anything as compiler args and log a warning that I was tired if seeing 😅

File: build.gradle (hunk 1 above)
Comment (makosblade) [reply build.gradle:54]: Oh, looks like they're already in the jvmArgs segment

**Review by geekbeast — APPROVED** [review]

**Pull request merged**